src/libsyntax/codemap.rs

```diff
@@ -129,6 +132,15 @@ pub struct Span {
     pub expn_id: ExpnId
 }
 
+/// Spans are converted to MultiSpans just before error reporting, either automatically,
+/// generated by line grouping, or manually constructed.
+/// In the latter case care should be taken to ensure that spans are ordered, disjoint,
+/// and point into the same FileMap.
+#[derive(Clone)]
+pub struct MultiSpan {
+    pub spans: SmallVector<Span>
```

Comment:
Yeah, I don't think perf in the error case is super-important, so just using Vec is probably better (and save all the changes to SmallVec, which I would otherwise say need some tests).